nbtlib/tag.py

```diff
@@ -477,8 +484,39 @@ def write(self, buff, byteorder='big'):
         for elem in self:
             elem.write(buff, byteorder)
 
+    def match(self, other):

```

Comment:
that is not how List's matching........

see https://github.com/vberlier/nbtlib/pull/35/files#diff-883b92497ef7de75b43f871ae381a365R25
for more.

...
```
    elif i is List:
        if pattern:
            return all(any(match(k,j) for k in obj) for j in pattern)
        else:
            return not obj
```
...

Reply:
and you forgot to make sure `other` is a List too.

Reply:
Oh okay interesting, so if the pattern is empty the list has to be empty and otherwise all the elements in the pattern must be matched but the order doesn't matter. I'm going to fix that, thanks!

Also the method only needs `other` to be iterable, so I feel like requiring an instance of `List` would make it unnecessarily restrictive.

```python
# No need to require this...
some_list_tag.match(List([Int(0), Int(1)]))

# ...when this works just fine!
some_list_tag.match([Int(0), Int(1)])
```

Reply:
Okay this should be fixed a3dfa21